### ISSUE TITLE
[Heron-ui]  Download of log files doesn't work

### ISSUE THREAD

**Issue by maosongfu** [twitter/heron#2307]
Opening the log file works, but download link there doesn't work. There are some cases where we can't scroll up all the way and search on the log page itself, and the only way is to download the file but the download link doesn't work:

The downloaded file doesn't contain data, it contains only an error message:
```{"status": "failure", "executiontime": 0.0030930042266845703, "message": "HTTP 599: Stream closed", "version": "0.14.11"}```

**Comment by maosongfu** [comment]
@cckellogg @objmagic 
Checked the log, it seems the heron-shell's `DownloadHandler` fails to respond the request from heron-tracker, and throws exceptions :
```
[E 170915 05:35:14 http1connection:53] Uncaught exception
    Traceback (most recent call last):
      File "/var/lib/mesos/slaves/e9952b74-f26a-412e-9e22-4dbcd96c1f17-S5987/frameworks/201104070004-0000002563-0000/executors/thermos-adreview-service-prod-adreview_cn_processor-5-90d3c247-8d44-4af1-a8a0-1efa3f44332c/runs/39a9d3fd-8b8d-49ac-9799-c0c33f32a9b6/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.ac906166e1f6becd21c962ec755200af39fec42d/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/http1connection.py", line 234, in _read_message
        delegate.finish()
      File "/var/lib/mesos/slaves/e9952b74-f26a-412e-9e22-4dbcd96c1f17-S5987/frameworks/201104070004-0000002563-0000/executors/thermos-adreview-service-prod-adreview_cn_processor-5-90d3c247-8d44-4af1-a8a0-1efa3f44332c/runs/39a9d3fd-8b8d-49ac-9799-c0c33f32a9b6/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.ac906166e1f6becd21c962ec755200af39fec42d/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/httpserver.py", line 282, in finish
        self.delegate.finish()
      File "/var/lib/mesos/slaves/e9952b74-f26a-412e-9e22-4dbcd96c1f17-S5987/frameworks/201104070004-0000002563-0000/executors/thermos-adreview-service-prod-adreview_cn_processor-5-90d3c247-8d44-4af1-a8a0-1efa3f44332c/runs/39a9d3fd-8b8d-49ac-9799-c0c33f32a9b6/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.ac906166e1f6becd21c962ec755200af39fec42d/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/web.py", line 1884, in finish
        self.execute()
      File "/var/lib/mesos/slaves/e9952b74-f26a-412e-9e22-4dbcd96c1f17-S5987/frameworks/201104070004-0000002563-0000/executors/thermos-adreview-service-prod-adreview_cn_processor-5-90d3c247-8d44-4af1-a8a0-1efa3f44332c/runs/39a9d3fd-8b8d-49ac-9799-c0c33f32a9b6/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.ac906166e1f6becd21c962ec755200af39fec42d/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/web.py", line 1904, in execute
        **self.handler_kwargs)
      File "/var/lib/mesos/slaves/e9952b74-f26a-412e-9e22-4dbcd96c1f17-S5987/frameworks/201104070004-0000002563-0000/executors/thermos-adreview-service-prod-adreview_cn_processor-5-90d3c247-8d44-4af1-a8a0-1efa3f44332c/runs/39a9d3fd-8b8d-49ac-9799-c0c33f32a9b6/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.ac906166e1f6becd21c962ec755200af39fec42d/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/web.py", line 180, in __init__
        self.initialize(**kwargs)
    TypeError: initialize() got an unexpected keyword argument 'path'
```

**Comment by objmagic** [comment]
resolved in latest Heron topologies (>0.15.0)
